app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt

```diff
@@ -34,64 +51,124 @@ class PoiInfoFragment(val poi: PointOfInterest) : Fragment() {
 
         // Places API query fields
         val placeFields = listOf(
-            Place.Field.ADDRESS,
-            Place.Field.PHONE_NUMBER,
-            Place.Field.WEBSITE_URI,
-            Place.Field.UTC_OFFSET,
-            Place.Field.OPENING_HOURS,
-            Place.Field.PHOTO_METADATAS
+                Place.Field.ADDRESS,
+                Place.Field.PHONE_NUMBER,
+                Place.Field.WEBSITE_URI,
+                Place.Field.UTC_OFFSET,
+                Place.Field.OPENING_HOURS,
+                Place.Field.PHOTO_METADATAS
         )
 
+        takeMeThereButton = view.findViewById(R.id.take_me_there_button)
+        takeMeThereButton.visibility = GONE
+
+        view.findViewById<TextView>(R.id.poi_name).text = poi.name
+
         val request = FetchPlaceRequest.newInstance(placeId, placeFields)
 
         placesClient.fetchPlace(request)
-            .addOnSuccessListener(getOnSuccessListener(view, placesClient)).addOnFailureListener {
-                val infoTextView = view.findViewById<TextView>(R.id.infoTextView)
-                infoTextView.text = "No information found for this point of interest :("
-            }
+                .addOnSuccessListener(getOnSuccessListener(view, placesClient)).addOnFailureListener {
+                    val infoTextView = view.findViewById<TextView>(R.id.infoTextView)
+                    infoTextView.text = "No information found for this point of interest :("
+                }
     }
 
     private fun getOnSuccessListener(
-        view: View,
-        placesClient: PlacesClient
+            view: View,
+            placesClient: PlacesClient
     ): (FetchPlaceResponse) -> Unit =
-        { response: FetchPlaceResponse ->
-            val place = response.place
+            { response: FetchPlaceResponse ->
+                val place = response.place
 
-            val openText = when {
-                place.isOpen == null -> ""
-                place.isOpen!! -> "OPEN"
-                else -> "CLOSED"
-            }
+                val openText = when {
+                    place.isOpen == null -> "No Info"
+                    place.isOpen!! -> "OPEN"
+                    else -> "CLOSED"
+                }
 
-            val infoTextView = view.findViewById<TextView>(R.id.infoTextView)
-            "${place.address}\n${place.phoneNumber}\n${place.websiteUri}\n${openText}".also {
-                infoTextView.text = it
-            }
+                val openStatusInfo = view.findViewById<TextView>(R.id.openStatusTextView)
+                openStatusInfo.text = openText
+
+                val infoTextView = view.findViewById<TextView>(R.id.infoTextView)
+                "${place.address}\n".also {
+                    infoTextView.text = it
+                }
+
+                if (place.websiteUri != null && place.websiteUri.toString().isNotEmpty()) {
+                    val richLinkView = view.findViewById<RichLinkViewSkype>(R.id.richLinkView)
```

Comment:
This is not working very well on my phone, on some pois it just shows as a white rectangle

Reply:
Yeah cause some websites don't allow the preview option. But you can always click and you'll get redirected to the website. If we have better ideas I propose we add them for next weeks task. What do you think?